docs/test-approach.md

```diff
@@ -0,0 +1,152 @@
+# Flexion Testing Approach
```

Comment:
@ericsorenson @mariahkannenberg have you reviewed @adunkman's feedback regarding this document? I wouldn't block accepting this PR on changes made to this doc, but would like to see updates per his feedback!